compiler/resolution/implementForallIntents.cpp

```diff
@@ -1723,6 +1724,105 @@ void implementForallIntents2wrapper(CallExpr* call, CallExpr* eflopiHelper)
 // for the (new) ForallStmt-based representation.
 //
 
+// Forall Intent information about each outer variable.
+class FIinfo {
+public:
+  Symbol* ovar;       // outer variable
+  Symbol* svar;       // shadow variable
+  Symbol* reduceGVar; // reduction class, non-NULL only for reduce intents
+  Symbol* reduceMark; // transitional measure; TODO: remove
+  TFITag  intent;     // the forall intent for this variable
+
+  bool isReduce() { return intent == TFI_REDUCE; }
+
+  FIinfo(Symbol* outerVar):
+    ovar(outerVar), svar(NULL),
+    reduceGVar(NULL), reduceMark(NULL), intent(TFI_DEFAULT)
+  { }
+};
+
+/*
+We need to:
+  * Associate each outer variable with a unique FIinfo.
+  * Adjust FIinfo with user-specified intent for each variable in with-clause.
+  * Iterate over the variables in a deterministic order.
+
+We can use just std::map<Symbol*,FIinfo>.
+
+Or we can use std::vector<FIinfo>.
+Also, to ensure uniqueness and for processing the with-clause,
+use a temporary std::map.
+
+Given my fuzzy-warm feeling about traversing a vector, i.e. faster than
+traversing a map, I chose to use the vector + temporary map.
+*/
+
+typedef std::vector<FIinfo> FIIvec;
+typedef std::map<Symbol*,int> FIImap;  // map to the index into FIIvec
+
+// Add 'sym' to 'fivec' and 'fimap', if it is not there already.
+static void addFII(Symbol* sym, FIIvec& fivec, FIImap& fimap) {
+  FIinfo info(sym);
+
+  if (fimap.count(sym))
+    // already there
+    return;
+
+  fivec.push_back(info);
+  int newIndex = (int)fivec.size()-1;
+  fimap.insert(std::make_pair(sym,newIndex));
+}
+
+/////////////////////////////////////////////////////////////////////////////
+// TFITag <-> IntentTag
+
+static IntentTag tfi2it(TFITag tfi) {
```

Comment:
Can you name this function in English? like getArgIntentForTaskIntent or something?